agent/test/start_scripts_test.py

```diff
@@ -6,17 +6,26 @@
 
 
 def test_startApiServer():
-    run_script('startApiServer')
+    port_range_end = 8090 # XXX
+    for port in range(8080, port_range_end):
```

Comment:
Somehow it feels lots or little confusing, this is what I can think of (there might be still better version of it which you may wanna try):

for port in range(8080, port_range_end):
	res = run_script('startApiServer', '0.0.0.0', str(port))
	if res == 2:  # 2 means address in use
		continue
	else:
		break
	
assert res == 0, 'script failed'
return